application/classes/Ushahidi/Repository/Dataprovider.php

```diff
@@ -28,6 +28,25 @@ class Ushahidi_Repository_Dataprovider implements
 	// ReadRepository
 	public function getEntity(Array $data = null)
 	{
+		// translating labels and description for data-providers
+		if($data !== null && $data['options']) {
+			foreach ($data['options'] as $option_key => $option) {
```

Comment:
I think you can use `&$option` so that modifications to `$option` stick on `$data`

Reply:
Yes, that worked, so much nicer and cleaner!